playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml

```diff
@@ -46,10 +46,36 @@
     - l_upgrade_nodes_drain_result is failed
     - openshift_upgrade_nodes_drain_timeout | default(0) | int == 0
 
+  # Run the pre-upgrade hook if defined:
+  - debug: msg="Running node pre-upgrade hook {{ openshift_node_upgrade_pre_hook }}"
+    when: openshift_node_upgrade_pre_hook is defined
+
```

Comment:
would be nicer if we use the asset module?

Reply:
How would you suggest the assert module is used here? Do you mean asserting that the referenced task file exists?